internal/config/yaml_parser.go

```diff
@@ -72,7 +76,16 @@ func IsValidLogRotateConfig(config LogRotateConfig) error {
 
 func (fileCacheConfig *FileCacheConfig) validate() error {
 	if fileCacheConfig.MaxSizeMB < -1 {
-		return fmt.Errorf("the value of max-size-mb for file-cache can't be less than -1")
+		return fmt.Errorf(FileCacheMaxSizeMBInvalidValueError)
+	}
+	if fileCacheConfig.MaxDownloadParallelism < -1 {
+		return fmt.Errorf(MaxDownloadParallelismInvalidValueError)
+	}
+	if fileCacheConfig.EnableParallelDownloads && fileCacheConfig.DownloadParallelismPerFile < 1 {
+		return fmt.Errorf(DownloadParallelismPerFileInvalidValueError)
+	}
+	if fileCacheConfig.EnableParallelDownloads && fileCacheConfig.ReadRequestSizeMB < 1 {
```

Comment:
As discussed offline, if we can remove the fileCacheConfig.EnableParallelDownloads from here.